Make nemesis restartable

kbr- opened the pull request:
The start_nemesis() function now clears the Event responsible for
telling nemesis threads to stop.

`BaseScyllaCluster.termination_event` was renamed to `BaseScyllaCluster.nemesis_termination_event` since it's only being used to notify nemesis threads to terminate.

Fixes #2724.

## PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

bentsi reviewed (changes requested):
As @fruch suggested: 
- add `self.termination_event.clear()` in `start_nemesis`.
- better to rename the `self.termination_event` to `self.nemesis_termination_event` since that's what it is used for

kbr- commented:
@bentsi as explained in the commit message, I assumed that `termination_event` is used for terminating everything, and added a new event, `nemesis_termination_tx` for terminating the nemesis only. This is to make code future-proof (someone might want to use `termination_event`).

I also added `self.nemesis_termination_tx.clear()` in `start_nemesis` as requested.

I can remove `self.termination_event` and leave only the nemesis-specific event. If someone wants to have an event for terminating everything in the future they'll have to add it back. Do you want me to do that?

kbr- commented:
Btw. I'm currently running a test which uses the new mechanism:
https://jenkins.scylladb.com/job/scylla-staging/job/abykov/job/feature-cdc-replication-longevity/14/console
as we can see in the latest logs, there are disruptions happening in the second round. Yay!

bentsi commented:
@kbr- I understand that... your code is cool but we don't need such complexity to solve this problem. 
There is a lot of legacy code here in SCT and naming was done bad...

kbr- commented:
Ok, I removed the new logic and simply use a single `nemesis_termination_event`.

bentsi reviewed (changes requested):
Please check following:
![image](https://user-images.githubusercontent.com/5269241/94560599-8faa7000-026b-11eb-95d0-f16f0569514b.png)

kbr- commented:
Fixed. Also updated PR description.

fruch reviewed (approved):
LGTM
